app/models/vote.rb

```diff
@@ -0,0 +1,7 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  belongs_to :work
+
+  validates :user_id, presence: true, uniqueness: { scope: :work, message: "You can only add one vote per work"}
```

Comment:
Since a vote belongs to a user by default `user_id` is required.  The uniqueness of combining `user_id `and `work_id` is good.  The custom message is great!